lab4/provided/record_store.cc

```diff
@@ -0,0 +1,73 @@
+#include "record_store.hh"
+
+#include <fstream>
+#include <sstream>
+
+istream& operator>>(istream& is, Record& r) {
+    is >> r.ticksFromNow >> r.write >> hex >> r.address >> dec >> r.requestId >> r.size;
+
+    r.dataVec.clear();
+    r.dataVec.reserve(r.size);
+    if (r.write) {
+        for (int i = 0; i < r.size; ++i) {
+            int byte;
+            is >> hex >> byte >> dec;
+            r.dataVec.push_back((uint8_t) byte);
+        }
+    }
+
+    return is;
+}
+
+ostream& operator<<(ostream& os, Record& r) {
+    os << r.ticksFromNow << ' ' << r.write << ' ' << hex << "0x" << r.address << dec << ' ' << r.requestId << ' ' << r.size; 
+
+    if (r.write) {
+        for (int i = 0; i < r.size; ++i) {
+            os << ' ';
+            os << hex << "0x" << int(r.dataVec[i]) << dec;
+        }
+    }
+
+    os << endl;
+
+    return os;
+}
+
+RecordStore::RecordStore(string filename):
+    filename(filename)
+{
+    //
```

Comment:
I guess this is something your editor does? I haven't seen an empty comment in the place of an empty function before. I don't care if you fix or not.

Reply:
That's actually just my own coding habit. I'll get rid of it.